File: Queue/nodejs/queue_class.js
// Queue class
class Queue {
    // Array is used to implement Queue
    constructor() {
      this.items = [];
    }

    enqueue(data){
        if(this.items.indexOf(data) == -1){
            this.items.push(data)
            return true;
        }
        return false 
    }

    dequeue(){
        if(this.items.length == 0){ 
            return false;
        }
        return this.items.splice(this.items.length-1, 1) 
    }

    peek(){
        if(this.items.length > 0){
            return this.items[this.items.length-1]
        }
    }

    isFull(size){
        return this.items.length === size
    }

    isEmpty(){
        return this.items.length === 0
    }

    size(){
        return this.items.length;
    }

    printQueueElements(){
        return this.items;
    }

  
}
 
let newQueue = new Queue()

console.log(newQueue.isEmpty())

newQueue.enqueue("First Value")
newQueue.enqueue("Second Value")
newQueue.enqueue("Third Value")
newQueue.enqueue("Fourth Value")


console.log(newQueue.printQueueElements())

console.log("Peek = ",newQueue.peek())

console.log("Dequeue Element > ",newQueue.dequeue())

console.log("Dequeue Element > ",newQueue.dequeue()) 

console.log(newQueue.isFull(3))

File: Queue/python/queue_class.py
class Queue:

    def __init__(self):
        self.queue = list()

    
    def enqueue(self, data):
        if data not in self.queue:
            self.queue.insert(0, data)
            return True
        return False

    def dequeue(self):
        if len(self.queue) == 0: 
            return False

        val =self.queue[0];
        self.queue.remove(val)
        return val

    def size(self):
        return len(self.queue)

    def peek(self):
        if len(self.queue) == 0: 
            return False
        return  self.queue[0]

    def isFull(self, size):
        return  len(self.queue) == size

    def printQueueElements(self): 
        elements= ''
        if len(self.queue) == 0:
            return 'Queue is empty' 
        return (self.queue)
 

#initialize the queue built based on list properties
newQueue = Queue() 

# insert operation
newQueue.enqueue("First Value")

newQueue.enqueue("Second Value")

newQueue.enqueue("Third Value")

newQueue.enqueue("Fourth Value")

# print elements
print(newQueue.printQueueElements())

# examine the queue size
print(newQueue.size())

#remove elements - pop operation on FIFO queue
print("dequeue 1=> ",newQueue.dequeue())
print("dequeue 2=> ",newQueue.dequeue())

# get the last added Item
print("Peek = ",newQueue.peek())
 
print(newQueue.isFull(3))
